src/wallet/wallet_tx_builder.h

```diff
@@ -174,7 +180,9 @@ class TransactionEffects {
             fee(fee),
             internalOVK(internalOVK),
             externalOVK(externalOVK),
-            anchorHeight(anchorHeight) {}
+            anchorHeight(anchorHeight) {
+        LockSpendable();
```

Comment:
Is putting this call in the constructor the right idea? In a future where we persist information about a proposed transaction across multiple RPC calls, this might make things harder, though I guess if locking is idempotent it's fine.

Reply:
Yeah, I moved it out of there. I had it outside there originally, then moved it in, but wasn’t a big fan of that. And then the explicit `wallet` passing made me move it out again.

And yeah, locking is idempotent. Just inserts things into a set. It the selection part that checks if locks are held, so re-locking won’t block or fail.